### DISCUSSION TITLE
can't refine on multiple values

### DISCUSSION THREAD

**Discussion by camilabellis** [algolia/instantsearch#6280] [Q&A]
In a multi-select useMenu, refinement cannot be applied to every selected value. Therefore, values that are not part of the search result are shown.
<img width="249" alt="image" src="https://github.com/algolia/instantsearch/assets/38572266/3075e62c-cd71-4438-831d-af969cbb970e">
<img width="245" alt="image" src="https://github.com/algolia/instantsearch/assets/38572266/3ba36a2a-cb5e-4727-b580-7bcef580e405">

In this example, the useMenu was refined with the 'German' value. All the other filters updated correctly. However, once 'English' is selected, the refinement resets, and the filters show the results for 'English' instead of 'German' + 'English', even though the search result has one hit (nbHits: 1).

**Comment by Haroenv** [comment]
`useMenu` is for single-select, if you want multiple selections, use `useRefinementList`

**Reply by camilabellis** [reply]
thanks for the answer! I tried it but I'm still getting the wrong options in the other filters, even though more than one option is refined

**Reply by Haroenv** [reply]
do you have a codesandbox that shows how it doesn't behave how you expect?
